src/gui/sharedialog.cpp

```diff
@@ -0,0 +1,308 @@
+    url.setQueryItems(getParams);
+    postData.setQueryItems(postParams);
+    OcsShareJob *job = new OcsShareJob("PUT", url, postData, AccountManager::instance()->account(), this);
+    connect(job, SIGNAL(jobFinished(QString)), this, SLOT(slotPasswordSet(QString)));
+    job->start();
+}
+
+void ShareDialog::slotPasswordSet(QString reply)
+{
+    _ui->labelPasswordSpinner->hide();
+}
+
+void ShareDialog::getShares()
+{
+    this->setWindowTitle(QString("Sharing ").append(_path));
+    QUrl url = Account::concatUrlPath(AccountManager::instance()->account()->url(), QLatin1String("ocs/v1.php/apps/files_sharing/api/v1/shares"));
+    QList<QPair<QString, QString> > params;
+    params.append(qMakePair(QString::fromLatin1("format"), QString::fromLatin1("json")));
+    params.append(qMakePair(QString::fromLatin1("path"), _path));
+    url.setQueryItems(params);
+    OcsShareJob *job = new OcsShareJob("GET", url, QUrl(), AccountManager::instance()->account(), this);
+    connect(job, SIGNAL(jobFinished(QString)), this, SLOT(slotSharesFetched(QString)));
+    job->start();
+}
+
+void ShareDialog::slotSharesFetched(QString reply)
+{
+    bool success = false;
+    QVariantMap json = QtJson::parse(reply, success).toMap();
+    ShareDialog::_shares = json.value("ocs").toMap().values("data")[0].toList();
+    for(int i = 0; i < ShareDialog::_shares.count(); i++)
+    {
+        QVariantMap data = ShareDialog::_shares[i].toMap();
+
+        if (data.value("share_type").toInt() == SHARETYPE_PUBLIC)
+        {
+            _public_share_id = data.value("id").toULongLong();
+
+            _ui->lineEdit_shareLink->show();
+            _ui->pushButton_copy->show();
+            _ui->checkBox_password->show();
+            _ui->checkBox_expire->show();
+            _ui->pushButton_copy->show();
+            _ui->checkBox_shareLink->setChecked(true);
+
+            if (data.value("share_with").isValid())
+            {
+                _ui->checkBox_password->setChecked(true);
+                _ui->lineEdit_password->setText("********");
+                _ui->lineEdit_password->show();
+            }
+
+            if (data.value("expire_date").isValid())
+            {
+                _ui->calendar->setSelectedDate(QDate::fromString(data.value("expire_date").toString(), "dd-MM-yyyy"));
+                _ui->calendar->show();
+                _ui->checkBox_expire->setChecked(true);
+            }
+
+            char url[512];
+            sprintf(url, "https://home.azelphur.com/owncloud/public.php?service=files&t=%s", data.value("token").toString().toStdString().c_str());
+            _ui->lineEdit_shareLink->setText(url);
+        }
+    }
+}
+
+void ShareDialog::slotDeleteShareFetched(QString reply)
+{
+    _ui->labelShareSpinner->hide();
+    _ui->lineEdit_shareLink->hide();
+    _ui->pushButton_copy->hide();
+    _ui->lineEdit_password->hide();
+    _ui->checkBox_password->hide();
+    _ui->checkBox_expire->hide();
+    _ui->calendar->hide();
+}
```

Comment:
consider putting those widgets in a common widget and just `hide()` and `show()` that.